File: apps/web/src/types/events.ts
export type VenueOut = {
  id: number;
  name: string;
  slug: string;
  area?: string | null;
};

export type VenueDetailOut = {
  id: number;
  name: string;
  slug: string;
  area?: string | null;
  address?: string | null;
  website?: string | null;
  description?: string | null;
  description_markdown?: string | null;
  hero_image_path?: string | null;
  timezone?: string | null;
};

export type CategoryOut = {
  id: number;
  name: string;
  slug: string;
};

export type EventOut = {
  id: number;
  title: string;
  slug: string;
  description?: string | null;
  is_free: boolean;
  price_text?: string | null;
  external_url?: string | null;
  status: "scheduled" | "canceled" | string;
  categories: CategoryOut[];
};

export type EventOccurrenceOut = {
  id: number;
  start_datetime_utc: string; // ISO
  end_datetime_utc?: string | null;
  event: EventOut;
  venue: VenueOut | null;
  location_text?: string | null;
};

export type EventDetailOut = {
  event: EventOut;
  next_occurrence: EventOccurrenceOut;
  upcoming_occurrences: EventOccurrenceOut[];
  more_from_venue: EventOccurrenceOut[];
};

File: apps/api/app/schemas/events.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class VenueOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    area: str | None = None


class VenueDetailOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    area: str | None = None
    address: str | None = None
    website: str | None = None
    description: str | None = None
    hero_image_path: str | None = None
    timezone: str | None = None


class CategoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str


class EventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    slug: str
    description: str | None = None
    is_free: bool
    price_text: str | None = None
    external_url: str | None = None
    status: str
    categories: list[CategoryOut] = Field(default_factory=list)


class EventOccurrenceOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    start_datetime_utc: datetime
    end_datetime_utc: datetime | None = None

    location_text: str | None = None
    venue: VenueOut | None = None

    event: EventOut


class EventCountOut(BaseModel):
    count: int
    start: date
    end: date
    timezone: str = "America/New_York"


class EventDetailOut(BaseModel):
    event: EventOut
    next_occurrence: EventOccurrenceOut
    upcoming_occurrences: list[EventOccurrenceOut] = Field(default_factory=list)
    more_from_venue: list[EventOccurrenceOut] = Field(default_factory=list)


class EventSlugResolutionOut(BaseModel):
    event_id: int
    canonical_segment: str
    is_unique: bool
